aiida/restapi/resources.py

```diff
@@ -306,11 +309,12 @@ def post(self):
                     self.trans.__label__ = tag
                     data.update(self.trans.get_formatted_result(tag))
 
-        # Remove all 'full_type's
+        # Remove 'full_type's when they're `None`
         for tag, entities in list(data.items()):
             updated_entities = []
             for entity in entities:
-                entity.pop('full_type', None)
+                if entity.get('full_type') is None:
+                    entity.pop('full_type', None)
```

Comment:
The `None` isn't really necessary here; no need to change though.

Reply:
Correction: it is necessary ;-)

Reply:
Well, if `full_type` is not present, it's needed, otherwise it will raise. On the other hand, if you mean the line above, then technically, yes. The value of `full_type` will/should never be an empty string, so no need to test whether it's `None`, however, for this particular code part, I like to be as explicit as possible.